File: CSharp/ByTier/B/2120_B_Square_Pool.cs
// Problem: CF 2120 B - Square Pool
// https://codeforces.com/contest/2120/problem/B

/*
B. Square Pool
Algorithms/Techniques: Simulation with mathematical analysis of ball trajectories in a square pool.

Time Complexity: O(n) per test case, where n is the number of balls.
Space Complexity: O(1), only using a constant amount of extra space regardless of input size.

The problem simulates balls being shot from initial positions (xi, yi) in directions (dx, dy) at speed 10^100 units/sec.
Since the balls bounce elastically off walls and the table is square, the path can be thought of as a straight line
in an extended grid (toroidal space). The ball ends up in a pocket if its trajectory intersects any corner,
i.e., when both coordinates become multiples of s after accounting for direction.

For each ball:
- If dx == dy, then x + y = k*s for some integer k => (x + y) % s == 0.
- Else dx != dy, then x - y = k*s for some integer k => (x - y) % s == 0.
Thus we check the modular condition to see whether the ball will end up in a pocket.
*/

using System;
using System.Collections.Generic;
using System.Linq;
using System.IO;
using System.Text;
using ll = System.Int64;

class CP {
  static void Main() {
    var input = new FastReader();
    var print =
        new StreamWriter(new BufferedStream(Console.OpenStandardOutput()));
    int t1 = input.readint();
    int mod = 1000000007;
    while (t1-- > 0) {
      int n = input.readint();
      int m = input.readint(); // m represents side length s of the square table
      int ans = 0;
      for (int i = 0; i < n; i++) {
        ll dx = input.readint();
        ll dy = input.readint();
        ll x = input.readint();
        ll y = input.readint();
        if (dx == dy) {
          // Ball moves along the line y = x + c => x - y = -c.
          // When it hits a pocket, it means x + y is divisible by m (s).
          if ((x - y) % m == 0)
            ans += 1;
        } else {
          // Ball moves along the line y = -x + c => x + y = c.
          // When x + y is divisible by s, it ends up in a pocket.
          if ((x + y) % m == 0)
            ans += 1;
        }
      }
      print.WriteLine(ans);
    }
    print.Flush();
  }
}

class FastReader {
  private readonly Stream s;
  private readonly byte[] b = new byte[1 << 18];
  private int l, p;
  public FastReader() { s = Console.OpenStandardInput(); }
  private byte Read() {
    if (p >= l) {
      p = 0;
      l = s.Read(b, 0, b.Length);
      if (l == 0)
        return 0;
    }
    return b[p++];
  }
  private byte Skip() {
    byte c;
    while ((c = Read()) <= 32)
      ;
    return c;
  }
  public int readint() {
    int n = 0, sgn = 1;
    byte c = Skip();
    if (c == '-') {
      sgn = -1;
      c = Read();
    }
    while (c >= '0' && c <= '9') {
      n = n * 10 + (c - '0');
      c = Read();
    }
    return n * sgn;
  }
  public long readlong() {
    long n = 0, sgn = 1;
    byte c = Skip();
    if (c == '-') {
      sgn = -1;
      c = Read();
    }
    while (c >= '0' && c <= '9') {
      n = n * 10 + (c - '0');
      c = Read();
    }
    return n * sgn;
  }
  public string readstring() {
    var sb = new System.Text.StringBuilder();
    byte c = Skip();
    while (c > 32) {
      sb.Append((char)c);
      c = Read();
    }
    return sb.ToString();
  }
  public int[] IntArray(int n) {
    var a = new int[n];
    for (int i = 0; i < n; i++)
      a[i] = readint();
    return a;
  }
  public List<int> IntList(int n) {
    var list = new List<int>(n);
    for (int i = 0; i < n; i++)
      list.Add(readint());
    return list;
  }
}


// https://github.com/VaHiX/codeForces/

File: Java/ByRound/2120/2120_B_Square_Pool.java
// Problem: CF 2120 B - Square Pool
// https://codeforces.com/contest/2120/problem/B

/*
 * Problem: B. Square Pool
 * Purpose: Determine how many balls are potted in a square pool table with pockets at each corner.
 *          Each ball is launched from an initial position with a direction vector and moves at constant speed.
 *          The table has side length 's' and corners at (0,0), (0,s), (s,0), (s,s).
 *          Balls bounce off the walls elastically and are potted if they reach any pocket.
 *
 * Algorithm:
 * - For each ball, determine if the trajectory will lead it to a pocket.
 * - A ball is potted if either:
 *   1. It moves diagonally along the diagonal (x == y && dx == dy)
 *      which leads directly to (s,s) pocket
 *   2. It moves such that x + y == s and the direction vector changes sign (dx != dy),
 *      leading it to the (0,s) or (s,0) pocket.
 *
 * Time Complexity: O(n) per test case, where n is the number of balls.
 * Space Complexity: O(1), only using a constant amount of extra space.
 */

import java.util.*;
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int tc = sc.nextInt();
        while(tc-- != 0) {
            int n = sc.nextInt(); // number of balls
            int s = sc.nextInt(); // side length of the square pool table
            int ans = 0;
            for(int i = 0; i < n; i++) {
                int dx = sc.nextInt(); // direction vector along x-axis
                int dy = sc.nextInt(); // direction vector along y-axis
                int x = sc.nextInt(); // initial x-coordinate of the ball
                int y = sc.nextInt(); // initial y-coordinate of the ball
                // Check if the ball will be potted:
                // Case 1: Moving toward (s,s) via diagonal path
                // Case 2: Moving such that x + y = s, which leads to a corner pocket
                if((x == y && dx == dy) || (x + y == s && dx != dy)) ans++;
            }
            System.out.println(ans);
        }
    }
}


// https://github.com/VaHiX/CodeForces/